core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -21,6 +21,10 @@ private[chisel3] trait PropertyType[T] {
   /** Get the IR PropertyType for this PropertyType.
     */
   def getPropertyType: ir.PropertyType
+
+  /** Get the IR PropertyLit for this PropertyType.
+    */
+  def getPropertyLit(lit: T): ir.PropertyLit
 }
```

Comment:
This being part of `PropertyType` makes sense for the currently existing property types, but what is it going to look like for objects? Are we committing to supporting literals for arbitrary user-defined types?

This question is not blocking, just something we need to keep in mind as we continue to flesh out this API.

Reply:
Hmm that's a good point, maybe we don't want to commit to that. I could revert back to the initial version, or roll with this for now.

Reply:
I'm going to flip back to the original version for now and plan to merge this.